components/calendar/src/iso.rs

```diff
@@ -236,6 +238,10 @@ impl Date<Iso> {
     /// assert_eq!(date_iso.day_of_month().0, 2);
     /// ```
     pub fn try_new_iso_date(year: i32, month: u8, day: u8) -> Result<Date<Iso>, CalendarError> {
+        if month > 12 {
+            return Err(CalendarError::OutOfRange);
+        }
+
```

Comment:
Nit: This condition shouldn't be necessary because the following `if` should capture this case